# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/efivar/hotfix.patch.ia64
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- efivar-39/src/include/gcc.specs.vanilla	2025-07-06 22:23:00.183502958 +0200
+++ efivar-39/src/include/gcc.specs	2025-07-06 22:23:34.380000000 +0200
@@ -2,7 +2,7 @@
 + %{!r:%{!fpie:%{!fPIE:%{!fpic:%{!fPIC:%{!fno-pic:-fPIE}}}}}} -grecord-gcc-switches
 
 *self_spec:
-+ %{!shared:%{!static:%{!r:-pie}}} %{static:-Wl,-no-fatal-warnings -Wl,-static -static -Wl,-z,relro,-z,now} -grecord-gcc-switches
++ %{!shared:%{!static:%{!r:-pie}}} %{static:-Wl,-no-fatal-warnings -Wl,-static -static -Wl,-z,now} -grecord-gcc-switches
 
 *link:
-+ %{!static:--fatal-warnings} --no-undefined-version --no-allow-shlib-undefined -z now --build-id %{!static:%{!shared:-pie}} %{shared:-z relro} %{static:%<pie}
++ %{!static:--fatal-warnings} --no-undefined-version --no-allow-shlib-undefined -z now --build-id %{!static:%{!shared:-pie}} %{static:%<pie}
--- efivar-39/src/include/defaults.mk.vanilla	2025-07-06 22:23:41.507375049 +0200
+++ efivar-39/src/include/defaults.mk	2025-07-06 22:23:49.953326910 +0200
@@ -57,7 +57,6 @@
 		   -Wl,--no-undefined-version \
 		   -Wl,-z,now \
 		   -Wl,-z,muldefs \
-		   -Wl,-z,relro \
 		   -Wl,--fatal-warnings \
 		   $(call family,LDFLAGS) $(call family,CCLDFLAGS) \
 		   $(call pkg-config-ccldflags)
@@ -90,7 +89,7 @@
 HOST_CFLAGS ?= $(CFLAGS) $(call family,HOST_CFLAGS)
 override _HOST_CFLAGS := $(HOST_CFLAGS)
 override HOST_CFLAGS = $(_HOST_CFLAGS)
-HOST_LDFLAGS_CLANG ?= -Wl,--fatal-warnings,-z,relro
+HOST_LDFLAGS_CLANG ?= -Wl,--fatal-warnings
 HOST_LDFLAGS_GCC ?= -Wl,--no-undefined-version
 HOST_LDFLAGS ?=
 HOST_CCLDFLAGS ?=
